### PR TITLE
fix: keep Gemma 3n pipeline state rank-local

### PR DESCRIPTION
**Pull request by ttupper92618**
## Summary
- reindex Gemma 3n shared-KV cache dependencies after pipeline slicing
- slice rank-local per-layer input embeddings and projections for both quantized and unquantized MLX modules
- reject shard boundaries that would consume K/V owned by another rank instead of running incorrect attention

## Why
The configured-fleet release battery found a deterministic two-rank Gemma 3n warmup crash: a shared layer read an empty rotating cache because full-model indices survived pipeline slicing. The same drift also selected the wrong per-layer input blocks.

## Validation
- `uv run basedpyright`
- `uv run ruff check`
- `nix --extra-experimental-features "nix-command flakes" fmt`
- `uv run pytest` (2948 passed, 1 skipped, 228 deselected)
- focused regression covers unquantized and 4-bit quantized layouts plus unsafe shard rejection

### CODE REVIEW

File: src/skulk/worker/engines/mlx/auto_parallel.py [hunk 1]
```diff
@@ -612,6 +612,216 @@ class _Gemma4ArgsLike(Protocol):
...
+) -> None:
+    """Retain one contiguous output-feature range in a linear module."""
+    if isinstance(module, nn.QuantizedLinear):
+        quantized = cast(_QuantizedLinearParameters, cast(object, module))
+        quantized.weight = quantized.weight[start_feature:end_feature]
+        quantized.scales = quantized.scales[start_feature:end_feature]
+        if quantized.biases is not None:
+            quantized.biases = quantized.biases[start_feature:end_feature]
+        if "bias" in module:
+            quantized.bias = quantized.bias[start_feature:end_feature]
```
[the remark this answers is not in the document]
Comment (ttupper92618) [reply src/skulk/worker/engines/mlx/auto_parallel.py:713]: Severity 1 (informational): MLX `nn.Module` intentionally supports membership checks for registered parameters—its own `QuantizedLinear.__call__` and representation use `"bias" in self`. The focused regression exercises this exact quantized `bias=False` branch and passes, so there is no missing/None slice. No change warranted.
